crates/polars-io/src/parquet/write/writer.rs

```diff
@@ -121,12 +121,17 @@ where
         }
     }
 
-    /// Write the given DataFrame in the writer `W`. Returns the total size of the file.
-    pub fn finish(self, df: &mut DataFrame) -> PolarsResult<u64> {
+    /// Write the given DataFrame along with optional key value metadata in the writer `W`.
+    /// Returns the total size of the file.
+    pub fn finish(
+        self,
+        df: &mut DataFrame,
+        key_value_metadata: Option<Vec<KeyValue>>,
```

Comment:
NOTE: Instead of modifying the signature here (and, thus, deviating from other writers), we could also add this to the parquet writer (on second thought, this is probably a better idea).

Reply:
Ah I didn't find all the tests locally, modifying the `ParquetWriter` seems to quite obviously be the better choice. I'll make that change tomorrow.